bindgen/ir/comp.rs

```diff
@@ -595,42 +606,41 @@ where
         let bitfield_width = bitfield.bitfield_width().unwrap() as usize;
         let bitfield_layout =
             ctx.resolve_type(bitfield.ty()).layout(ctx).ok_or(())?;
-        let bitfield_size = bitfield_layout.size;
         let bitfield_align = bitfield_layout.align;
+        // bitfield offset of struct/union
+        let bitfield_offset = bitfield.offset().unwrap();
 
-        let mut offset = unit_size_in_bits;
-        if !packed {
-            if is_ms_struct {
-                if unit_size_in_bits != 0 &&
-                    (bitfield_width == 0 ||
-                        bitfield_width > unfilled_bits_in_unit)
-                {
-                    // We've reached the end of this allocation unit, so flush it
-                    // and its bitfields.
-                    unit_size_in_bits =
-                        align_to(unit_size_in_bits, unit_align * 8);
-                    flush_allocation_unit(
-                        fields,
-                        bitfield_unit_count,
-                        unit_size_in_bits,
-                        unit_align,
-                        mem::take(&mut bitfields_in_unit),
-                        packed,
-                    );
+        if unit_size_in_bits == 0 {
+            bitfields_bit_begin_off = bitfield_offset;
+        }
 
-                    // Now we're working on a fresh bitfield allocation unit, so reset
-                    // the current unit size and alignment.
-                    offset = 0;
-                    unit_align = 0;
-                }
-            } else if offset != 0 &&
+        if !packed &&
+            is_ms_struct &&
+            (unit_size_in_bits != 0 &&
                 (bitfield_width == 0 ||
-                    (offset & (bitfield_align * 8 - 1)) + bitfield_width >
-                        bitfield_size * 8)
-            {
-                offset = align_to(offset, bitfield_align * 8);
-            }
+                    bitfield_width > unfilled_bits_in_unit))
+        {
+            // We've reached the end of this allocation unit, so flush it
+            // and its bitfields.
+            unit_size_in_bits = align_to(unit_size_in_bits, unit_align * 8);
+            flush_allocation_unit(
+                fields,
+                bitfield_unit_count,
+                unit_size_in_bits,
+                unit_align,
+                mem::take(&mut bitfields_in_unit),
+                packed,
+            );
+
+            // Now we're working on a fresh bitfield allocation unit, so reset
+            // the current unit size and alignment.
+            unit_align = 0;
+            bitfields_bit_begin_off = bitfield_offset;
+            // uncomment when support msvc
+            // unit_size_in_bits = 0;
         }
+        // depended clang report field offset, manual calculates sometime not true
```

Comment:
Cannot make sense of this comment, can you reword?

I think you are saying something like:

```
// Use the clang-reported offset, since the manual calculation may be off.
```

Or so? Or the other way around?